tetris.py

```diff
@@ -407,89 +467,65 @@ def flasher(self, full_lines, fill):
+            self.game.pause_start = time.time()
 
     def move_block_left(self, event):
-        if self.game_status == GameStatus.in_progress:
-            self.main_board.user_input_left()
-            self.main_board.draw_block()
+        if self.game.game_status == GameStatus.in_progress:
+            self.game.user_input_left()
 
     def move_block_right(self, event):
-        if self.game_status == GameStatus.in_progress:
-            self.main_board.user_input_right()
-            self.main_board.draw_block()
-
-    def move_block_down(self):
-        if self.game_status == GameStatus.in_progress:
-            self.main_board.move_current_block_down()
-            self.game_over_check()
-            self.main_board.draw_block()
-        self.main_board.canvas.after(game_speed, self.move_block_down)
+        if self.game.game_status == GameStatus.in_progress:
+            self.game.user_input_right()
 
     def move_block_down_press(self, event):
-        if not self.main_board.fast_down:
-            self.main_board.fast_down = True
-            self.main_board.user_input_down()
+        if not self.game.fast_down:
+            self.game.fast_down = True
+            self.game.user_input_down()
 
     def move_block_down_release(self, event):
-        self.main_board.fast_down = False
+        self.game.fast_down = False
 
     def rotate_block(self, event):
-        if self.game_status == GameStatus.in_progress:
-            self.main_board.block_rotator()
-            self.main_board.draw_block()
-
-    def timer(self):
-        if self.game_status == GameStatus.in_progress:
-            game_time = time.time() - self.start_time - self.paused_time
-            self.topbar_time.config(text=f"{int(game_time / 60):02d}:{int(game_time % 60):02d}")
-            self.topbar_time.after(1000, self.timer)
+        if self.game.game_status == GameStatus.in_progress:
+            self.game.block_rotator()
 
-    def new_game(self):
-        self.small_board = Board(
-            self.topbar_canvas,
-            (1, 1),
-            None,
-            None,
-        )
-        self.small_board.resize_to_fit()
-        self.small_board.draw_block()
-
-        self.main_board = Board(
-            self.game_canvas,
-            (int(game_width / 2), -2),
-            self.small_board,
-            self.topbar_score,
-        )
-        self.start_time = time.time()
-        self.pause_start = 0
-        self.paused_time = 0
-        self.game_status = GameStatus.in_progress
-        self.main_board.game_score = 0
-        self.main_board.topbar_score.config(text=self.main_board.game_score)
+    def game_over(self, forced=False):
+        y_coordinates = [y for (x, y) in self.game.coord_extractor()]
+        if any(y < 0 for y in y_coordinates):
+            self.game.game_status = GameStatus.game_over
+        elif forced == True:
+            self.game.game_status = GameStatus.game_over
+            new_game()
```

Comment:
You never want to run `new_game()` without setting the status of the old game to `game_over`. Setting the status should be a part of `new_game()`, so that it's impossible to forget it.

Reply:
something like

```
if tetris_control.game is not None:
    tetris_control.game.status = GameStatus.game_over
```